rdkit/Chem/UnitTestRegistrationHash.py

```diff
@@ -0,0 +1,641 @@
+"""
+Tests for rdkit.Chem.RegistrationHash
+
+Focus is on molecules that have differing SMILES, but
+are actually the same.
+"""
+
+import textwrap
+import unittest
+
+from rdkit import Chem
+
+from rdkit.Chem import RegistrationHash
+from rdkit.Chem.RegistrationHash import HashLayer
+
+
+def hash_sdf(molblock: str, escape=None, data_field_names=None):
+    """
+    Gets the layers of the SDF, and generates the hash based on only the layers passed in
+
+    :param molblock: The molblock to hash
+    :param escape: optional field which can contain arbitrary information
+    :param data_field_names: optional sequence of names of SGroup DAT fields which
+       will be included in the hash.
+
+    :return: A dict with the hash & the layers
+    """
+    mol = Chem.MolFromMolBlock(molblock)
+    return RegistrationHash.GetMolLayers(mol,
+                                    escape=escape,
+                                    data_field_names=data_field_names)
+
+
+class CanonicalizerTest(unittest.TestCase):
+    maxDiff = 2000 # hash diffs can be long!
+
+    def test_example_structure(self):
+
+        structure = """
```

Comment:
I really dislike this way of including mol blocks since it makes it impossible to copy the mol block out and paste it somewhere else to visualize the molecule (and this is normally the first thing I do when a test is failing).

I won't insist if you feel strongly that the incremental readability improvements of the testing code are somehow more important than that, but please do consider getting rid of this indentation

Reply:
updated! I do this because it lets my editor fold the tests, but I agree that probably isn't worth the obnoxious hassle of swapping these over to something pastable - especially since the trailing spaces get dropped!